pyresttest/validators.py

```diff
@@ -289,6 +290,24 @@ def parse(cls, config, extractor_base=None):
         return cls.configure_base(config, base)
 
 
+class CookieExtractor(AbstractExtractor):
+    """ Extractor that pulls out a named cookie """
+    extractor_type = 'cookie'
+    is_header_extractor = True
+
+    def extract_internal(self, query = None, args = None, body = None, headers = None):
+        extracted = [y[1] for y in filter(lambda x: x[0] == 'set-cookie', headers) if y[1].startswith(query + '=')]
+        if len(extracted) == 0:
+            raise ValueError("Cookie named {0} not known".format(query))
+        else:
+            return str(dict(Cookie.SimpleCookie(extracted[0]))[query].value)
```

Comment:
Good point on the casting, it is not needed. Just an artefact for one of my iterations. Without it I think it will read straight-forward, just looping and looking for the header value starting with the case-sensitive name matching query.  So casting is about to go away.

As for the multiple headers. RFC 6265 states **"Servers SHOULD NOT include more than one Set-Cookie header field in the same response with the same cookie-name."**. And the suggested way for agents to handle this is only accept the last Set-Cookie header sent for a given name/path. I did not see how to pass args to an extractor in the config files, I see it done in the tests though. If you can give me the syntax on that I think a **path** arg to the extractor would be the best way to handle that situation instead of a list.